synapse/lib/parser.py

```diff
@@ -175,72 +194,64 @@ def subquery(self, meta, kids):
 
         return kid
 
-    def _parseJsonToken(self, meta, tokn):
+    def _parseJsonToken(self, tokn):
+
         if isinstance(tokn, lark.lexer.Token) and tokn.type == 'VARTOKN' and not tokn.value[0] in ('"', "'"):
+
             valu = tokn.value
+            astinfo = self.metaToAstInfo(tokn)
+
             try:
                 valu = float(valu) if '.' in valu else int(valu, 0)
             except ValueError as e:
-                mesg = f"Unexpected unquoted string in JSON expression at line {meta.line} col {meta.column}"
-                raise s_exc.BadSyntax(mesg=mesg, at=meta.start_pos, line=meta.line, column=meta.column)
+                self.raiseBadSyntax('Unexpected unquoted string in JSON expression', astinfo)
 
-            return s_ast.Const(valu)
+            return s_ast.Const(astinfo, valu)
         else:
             return self._convert_child(tokn)
 
     @lark.v_args(meta=True)
     def exprlist(self, meta, kids):
-        kids = [self._parseJsonToken(meta, k) for k in kids]
-        node = s_ast.ExprList(kids=kids)
-        node.lines = (meta.line, meta.end_line)
-        node.textpos = (meta.start_pos, meta.end_pos)
-        return node
+        kids = [self._parseJsonToken(k) for k in kids]
+        astinfo = self.metaToAstInfo(meta)
+        return s_ast.ExprList(astinfo, kids=kids)
 
     @lark.v_args(meta=True)
     def exprdict(self, meta, kids):
-        kids = [self._parseJsonToken(meta, k) for k in kids]
-        node = s_ast.ExprDict(kids=kids)
-        node.lines = (meta.line, meta.end_line)
-        node.textpos = (meta.start_pos, meta.end_pos)
-        return node
+        kids = [self._parseJsonToken(k) for k in kids]
+        astinfo = self.metaToAstInfo(meta)
+        return s_ast.ExprDict(astinfo, kids=kids)
 
     @lark.v_args(meta=True)
     def trycatch(self, meta, kids):
         kids = self._convert_children(kids)
-        return s_ast.TryCatch(kids=kids)
+        astinfo = self.metaToAstInfo(meta)
```

Comment:
There's a bunch of these rules that can just be in the 1 to 1 conversion map, tossed up a PR cleaning them up here https://github.com/vertexproject/synapse/pull/3092